contracts/Template.sol

```diff
@@ -1,106 +1,162 @@
-        Voting voting = Voting(dao.newAppInstance(votingAppId, latestVersionAppBase(votingAppId)));
-        TokenManager tokenManager = TokenManager(dao.newAppInstance(tokenManagerAppId, latestVersionAppBase(tokenManagerAppId)));
-
-        MiniMeToken token = tokenFactory.createCloneToken(MiniMeToken(0), 0, "App token", 0, "APP", true);
-        token.changeController(tokenManager);
-
-        // Initialize apps
-        app.initialize();
-        tokenManager.initialize(token, true, 0);
-        voting.initialize(token, 50 * PCT, 20 * PCT, 1 days);
+    /**
+    * @dev Deploy a Template DAO using a previously cached MiniMe token
+    * @param _holders Array of token holder addresses
+    * @param _stakes Array of token stakes for holders (token has 18 decimals, multiply token amount `* 10^18`)
+    * @param _votingSettings Array of [supportRequired, minAcceptanceQuorum, voteDuration] to set up the voting app of the organization
+    */
+    function newInstance(
+        address[] memory _holders,
+        uint256[] memory _stakes,
+        uint64[3] memory _votingSettings
+    )
+        public
+    {
+        _ensureTemplateSettings(_holders, _stakes, _votingSettings);
+
+        (Kernel dao, ACL acl) = _createDAO();
+        (Voting voting) = _setupBaseApps(dao, acl, _holders, _stakes, _votingSettings);
+        _setupCustomApp(dao, acl, voting);
+        _transferRootPermissionsFromTemplateAndFinalizeDAO(dao, voting);
+    }
 
-        acl.createPermission(this, tokenManager, tokenManager.MINT_ROLE(), this);
-        tokenManager.mint(root, 1); // Give one token to root
+    function _setupBaseApps(
+        Kernel _dao,
+        ACL _acl,
+        address[] memory _holders,
+        uint256[] memory _stakes,
+        uint64[3] memory _votingSettings
+    )
+        internal
+        returns (Voting)
+    {
+        MiniMeToken token = _popTokenCache(msg.sender);
+        TokenManager tokenManager = _installTokenManagerApp(_dao, token, TOKEN_TRANSFERABLE, TOKEN_MAX_PER_ACCOUNT);
+        Voting voting = _installVotingApp(_dao, token, _votingSettings);
+
+        _mintTokens(_acl, tokenManager, _holders, _stakes);
+        _setupBasePermissions(_acl, voting, tokenManager);
+
+        return (voting);
+    }
 
-        acl.createPermission(ANY_ENTITY, voting, voting.CREATE_VOTES_ROLE(), root);
+    function _setupBasePermissions(
+        ACL _acl,
+        Voting _voting,
+        TokenManager _tokenManager
+    )
+        internal
+    {
+        _createEvmScriptsRegistryPermissions(_acl, _voting, _voting);
+        _createVotingPermissions(_acl, _voting, _voting, _tokenManager, _voting);
+        _createTokenManagerPermissions(_acl, _tokenManager, _voting, _voting);
+    }
 
-        acl.createPermission(voting, app, app.INCREMENT_ROLE(), voting);
-        acl.createPermission(ANY_ENTITY, app, app.DECREMENT_ROLE(), root);
-        acl.grantPermission(voting, tokenManager, tokenManager.MINT_ROLE());
+    function _setupCustomApp(
```

Comment:
Last thing (I promise!):

Let's add some comments around the custom app functions (and where they're used) to make it super obvious this is what users should edit to get their own app into the template.